react-with-json-logic/src/components/questionnaireEditor/formEditor/testCaseEditor/ElementEditorTestCase.tsx

```diff
@@ -0,0 +1,185 @@
+        onResultItemChange={onAnswerItemChange}
+      />
+      <Typography>Results</Typography>
+      <AnswerOrResultEditor
+        availableItems={availableResults}
+        currentValues={testCase.results}
+        onResultItemChange={onResultItemChange}
+      />
+      <TestCaseResult testResult={runOneTestCase(questionnaireJson, testCase)} />
+    </Grid>
+  );
+}
+
+const AnswerOrResultEditor: React.FC<{
+  availableItems: AvailableItems;
+  currentValues: { [id: string]: any };
+  onResultItemChange: OnItemChange;
+}> = ({ currentValues, availableItems, onResultItemChange }) => {
+  const useStyles = makeStyles(() => ({
+    ul: {
+      listStyleType: "none",
+    },
+  }));
+
+  const classes = useStyles();
+
+  const unusedItemIds = availableItems.map((it) => it.itemId).filter((it) => !Object.keys(currentValues).includes(it));
+  return (
+    <ul className={classes.ul}>
+      {Object.entries(currentValues).map(([id, value]) => (
+        <OneItem
+          currentId={id}
+          currentValue={value}
+          availableIds={[id, ...unusedItemIds]}
+          availableValues={availableItems.find((it) => it.itemId === id)?.possibleValues ?? []}
+          onResultItemChange={onResultItemChange}
+        />
+      ))}
+      <OneItem availableIds={unusedItemIds} availableValues={[]} onResultItemChange={onResultItemChange} />
+    </ul>
+  );
+};
+
+const OneItem: React.FC<{
+  currentId?: string;
+  currentValue?: any;
+  availableIds: string[];
+  availableValues: any[];
+  onResultItemChange: OnItemChange;
+}> = ({ currentId, currentValue, availableIds, availableValues, onResultItemChange }) => {
+  const useStyles = makeStyles((theme) => ({
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: 240,
+    },
+  }));
+
+  const classes = useStyles();
+
+  return (
+    <li>
+      {currentId !== undefined ? (
+        <FormControl className={classes.formControl}>
+          <InputLabel id={"label-value-" + currentId}>{currentId}</InputLabel>
+          <Select
+            labelId={"label-value-" + currentId}
+            id={"value-" + currentId}
+            value={currentValue}
+            onChange={(event) => onResultItemChange({ itemId: currentId, value: event.target.value as string })}
+          >
+            {availableValues.map((value) => (
+              <MenuItem value={value}>{value}</MenuItem>
+            ))}
+          </Select>
+        </FormControl>
+      ) : null}
+      {currentId === undefined && availableIds.length > 0 ? (
+        <FormControl className={classes.formControl}>
+          <InputLabel id={"label-new-itemId"}>Result Category</InputLabel>
+          <Select
+            labelId={"label-new-itemId"}
+            id={"new-itemId"}
+            onChange={(event) => onResultItemChange({ itemId: event.target.value as string, value: "" })}
+          >
+            {availableIds.map((itemId) => (
+              <MenuItem value={itemId}>{itemId}</MenuItem>
```

Comment:
Should we show the question text to select instead of the ID?

Reply:
In my opinion the question text is too long. I went through creating a testcase for the CovApp questionnaire a few times during development, and I think that a well-chosen ID is better suited than the long question-text. We could think about adding a number to each question (just in the editor), which appears in the menu and in the test cases.